src/DbDumper.php

```diff
@@ -279,7 +279,11 @@ protected function echoToFile(string $command, string $dumpFile): string
         if ($this->compressor) {
             $compressCommand = $this->compressor->useCommand();
 
-            return "(((({$command}; echo \$? >&3) | {$compressCommand} > {$dumpFile}) 3>&1) | (read x; exit \$x))";
+            if ($this->isWindows()) {
```

Comment:
Extract this to a function called `getCompressCommand`. Inside that function, avoid using an `if` but easy early returns.

Reply:
Done